packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -121,11 +121,21 @@ export interface SharedTreeBranchEvents<TEditor extends ChangeFamilyEditor, TCha
 	dispose(): void;
 }
 
+/**
+ * Events related to branch trimming. This is done as a performance optimization.
+ */
+export interface BranchTrimmingEvents {
+	/**
+	 * Fired when commits are trimmed from the branch.
+	 */
+	branchTrimmed(trimmedRevisions: RevisionTag[]): void;
+}
+
 /**
  * A branch of changes that can be applied to a SharedTree.
  */
 export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> extends EventEmitter<
-	SharedTreeBranchEvents<TEditor, TChange>
+	SharedTreeBranchEvents<TEditor, TChange> & BranchTrimmingEvents
```

Comment:
Do you think it'd be cleaner to just make `SharedTreeBranchEvents` extend `BranchTrimmingEvents`?